src/bartiq/compilation/_evaluate.py

```diff
@@ -119,7 +119,20 @@ def _parse_assignments(routine: Routine, assignments: list[str]) -> list[Assignm
     # Sort the assignments based on whether they refer to register size variables or not
     parsed_assignments: list[Assignment] = []
     for variable, value_str in assignment_map.items():
-        value = parse_value(value_str)
+        try:
+            value = parse_value(value_str)
+        except BartiqCompilationError:
```

Comment:
Overall this logic looks suspicious to me as there's non-trivial logic in `except` clause, which always raises my suspicion.
Unfortunately I don't have time to look into it right now, leaving this to @dexter2206.